### ISSUE TITLE
fails to import No-Intro Love Packs

### ISSUE THREAD

**Issue by bdeshi** [alucryd/oxyromon#113]
~Oxymoron~ 🤦‍♂️ Oxyromon fails to import some dats from the default [No-Intro Love Pack (Standard)](https://datomatic.no-intro.org/index.php?page=download&op=daily&s=107) archive.

Source dats:
- No-Intro/VTech - Mobigo (20220207-083241).dat
- Unofficial/Unofficial - Sony - PlayStation Vita (BlackFinPSV) (20240104-010459).dat
Error
```
Error: SimpleError { err: "Failed to parse DAT file, missing field `games`" }
```

Source dats:
- No-Intro/Nintendo - Wii U (Digital) (CDN) (20231116-082151).dat
- Redump Custom/Nintendo - Wii U - WUX (515) (2023-12-22).dat
- Redump Custom/Nintendo - GameCube - NKit RVZ [zstd-19-128k] (1984) (2023-12-22).dat
- Redump Custom/Nintendo - GameCube - NKit CISO [lossless] (1984) (2023-12-22).dat
- Unofficial/Unofficial - Video Game OSTs (Hardware Recordings) (20231224-223249).dat
Error:
```
Error: SimpleError { err: "Failed to parse DAT file, cannot parse integer from empty string" }
```
These files have some `size=""` fields. replacing this with `size="0"` passes. (Some also have no empty checksums but they don't seem to cause any parsing issues.)

---

Some of these may not be practical, but I think importing the no-intro archives as-is would be a good user experience.

**Comment by maxexcloo** [comment]
Adding to this, I get a hang on the "Nintendo - Nintendo DSi (Digital) (CDN) (Encrypted) (Parent-Clone) (20230417-043358).dat" DAT:

```
Processing "Nintendo - Nintendo DSi (Digital) (CDN) (Encrypted) (Parent-Clone) (20230417-043358).dat"
System: Nintendo - Nintendo DSi (Digital) (CDN) (Encrypted) (Parent-Clone)
Version: 20230417-043358
Games: 804
Processing system
Deleting old games
Processing games
```

**Comment by maxexcloo** [comment]
Another one that fails to import is the MAME DAT `DATs/ROMs_(machine+bios+device)/MAME_ROMs_262.dat` from https://www.progettosnaps.net//dats/MAME/packs/MAME_Dats_262.7z

```
Processing "MAME_ROMs_262.dat"
System: MAME - ROMs
Version: 0.262
Games: 41120
Processing system
Deleting old games
Processing games
⠤                                                                                                               40009/41120 22.5544/s 00:06:07 (00:00:49) [#############################################################################---]
thread 'main' panicked at src/import_dats.rs:469:49:
called `Option::unwrap()` on a `None` value
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
```

**Comment by alucryd** [comment]
Taken care of in https://github.com/alucryd/oxyromon/commit/9e2eefec55874beb379dc5ae75a52d0c3d9f1aa4.

@bdeshi I was able to import the whole love pack with this, roms with no size will simply be ignored (and a warning printed in the console).

@maxexcloo Also took care of the MAME dat, it looks a bit odd but I may not fully grasp it yet. I thought `romof` without a `cloneof` was used to signify a dependency to a bios, but here there are `romof` referring to non bios games (granted they are devices, and I added that info to the database in case it comes in handy in the future). Also the DAT has a few bogus entries, roms that are marked as `merge` are supposed to be present in either the parent game or the bios, but there are quite a few here that aren't, you'll see some warnings for them.

As for the hang, it happens sometimes here as well, I think it's just the sqlite database choking when there's a lot of activity, mostly during `import-dats` as it's quite database heavy. It's rare enough and unblocks itself after a few seconds though.
